django/db/backends/postgresql/base.py

```diff
@@ -149,6 +151,11 @@ def get_connection_params(self):
             raise ImproperlyConfigured(
                 "settings.DATABASES is improperly configured. "
                 "Please supply the NAME value.")
+        # PostgreSQL has a maximum length of 63 for identifiers.
+        if settings_dict['NAME'] is not None and len(settings_dict['NAME']) > 63:
```

Comment:
do you mean `if 'NAME' in settings_dict`?

Reply:
I think not. None may be used to connect to the default 'postgres' db. However, ```if 'NAME' in settings_dict``` prevents KeyError, but would still return True if 'NAME' is assigned a 'NoneType'. Referred ```test_default_name``` method in tests/backends/base/test_creation.py.